src/main/java/eu/bbmri_eric/negotiator/service/BBMRIDiscoveryServiceClientImpl.java

```diff
@@ -0,0 +1,244 @@
+import org.springframework.web.reactive.function.client.WebClientResponseException;
+
+@CommonsLog
+@Service(value = "DefaultDiscoveryServiceClient")
+public class BBMRIDiscoveryServiceClientImpl implements DiscoveryServiceClient {
+
+  // @Value("${negotiator.molgenis-url}")
+  // private String molgenisURL;
+
+  private BBMRIDiscoveryServiceClientImpl bbmriService = null;
+
+  @Autowired private OrganizationRepository organizationRepository;
+
+  @Autowired private AccessFormRepository accessFormRepository;
+
+  @Autowired private DiscoveryServiceRepository discoveryServiceRepository;
+  @Autowired private ResourceRepository resourceRepository;
+
+  @Autowired
+  private DiscoveryServiceSynchronizationJobRepository discoveryServiceSynchronizationJobRepository;
+
+  private DiscoveryServiceSynchronizationJob jobRecord;
+
+  private WebClient webClient;
+
+  @Autowired
+  public BBMRIDiscoveryServiceClientImpl(@Value("${negotiator.molgenis-url}") String molgenisURL) {
+    final int size = 16 * 1024 * 1024;
+    final ExchangeStrategies strategies =
+        ExchangeStrategies.builder()
+            .codecs(codecs -> codecs.defaultCodecs().maxInMemorySize(size))
+            .build();
+    this.webClient =
+        WebClient.builder().baseUrl(molgenisURL).exchangeStrategies(strategies).build();
+  }
+
+  public void syncAllOrganizations() {
+    List<MolgenisBiobank> biobanks = findAllBiobanks();
+    addMissingOrganizations(biobanks);
+  }
+
+  public void syncAllResources() {
+    List<MolgenisCollection> collections = findAllCollections();
+    addMissingResources(collections);
+  }
+
+  public void syncAllNetworks() {}
+
+  public void addMissingOrganizations(List<MolgenisBiobank> directoryBiobanks) {
+    for (MolgenisBiobank bb : directoryBiobanks) {
+      String biobankId = bb.getId();
+      Optional<Organization> organization = organizationRepository.findByExternalId(biobankId);
+      if (organization.isEmpty()) {
+        addMissingOrganization(bb);
+      } else {
+        log.debug(String.format("Biobank %s already present, check for updates...", biobankId));
+        if (!bb.getName().equals(organization.get().getName())) {
+          updateOrganizationName(organization.get(), bb);
+        }
+      }
+    }
+  }
+
+  public void addMissingResources(List<MolgenisCollection> directoryCollections) {
+    DiscoveryService discoveryService =
+        discoveryServiceRepository.findById(Long.valueOf("1")).get();
+    AccessForm accessForm = accessFormRepository.findById(Long.valueOf("1")).get();
+    for (MolgenisCollection coll : directoryCollections) {
+      Optional<Resource> resource = resourceRepository.findBySourceId(coll.getId());
+      if (resource.isEmpty()) {
+        addMissingResource(coll, discoveryService, accessForm);
+      } else {
+        log.debug(
+            String.format("Collection %s already present, check for updates...", coll.getId()));
+        if (!resource.get().getName().equals(coll.getName())
```

Comment:
why are all these methods public? The interface has way less methods

Reply:
Fixed in 53131252c6f43525c044b84ec3b7c262b19835ed. Was a big issue, now they are all private and also the tests have been refactored accordingly (they now call only the exposed methods)